pkg/pgmodel/cache/series_cache_test.go

```diff
@@ -52,3 +55,43 @@ func TestGenerateKey(t *testing.T) {
 	require.Equal(t, "test", metricName)
 	require.Equal(t, []byte("\x08\x00__name__\x04\x00test\x04\x00hell\x06\x00oworld\x05\x00hello\x05\x00world"), keyBuffer.Bytes())
 }
+
+func TestGrowSeriesCache(t *testing.T) {
+	testCases := []struct {
+		name         string
+		sleep        time.Duration
+		cacheGrowCnt int
+	}{
+		{
+			name:         "Grow criteria satisfied - we shoulnd't be evicting elements",
+			sleep:        time.Millisecond,
+			cacheGrowCnt: 1,
+		},
+		{
+			name:         "Growth criteria not satisfied - we can keep evicting old elements",
+			sleep:        time.Second * 2,
+			cacheGrowCnt: 0,
+		},
+	}
+
+	os.Setenv("IS_TEST", "true")
```

Comment:
```suggestion
	t.Setenv("IS_TEST", "true")
```

Reply:
It's better to use `t.Setenv` because it cleans the envvar after it finishes while `os.Setenv` will leave the variable set for subsequent tests.